build-tools/src/main/java/org/elasticsearch/gradle/testclusters/TestClustersAware.java

```diff
@@ -27,22 +26,27 @@ public interface TestClustersAware extends Task {
     @ServiceReference(REGISTRY_SERVICE_NAME)
     Property<TestClustersRegistry> getRegistery();
 
+    @ServiceReference(TEST_CLUSTER_TASKS_SERVICE)
+    Property<TestClustersPlugin.TaskEventsService> getTasksService();
+
     default void useCluster(ElasticsearchCluster cluster) {
         if (cluster.getPath().equals(getProject().getPath()) == false) {
             throw new TestClustersException("Task " + getPath() + " can't use test cluster from" + " another project " + cluster);
         }
 
         cluster.getNodes()
             .all(node -> node.getDistributions().forEach(distro -> dependsOn(getProject().provider(() -> distro.maybeFreeze()))));
-        cluster.getNodes().all(node -> dependsOn((Callable<Collection<Configuration>>) node::getPluginAndModuleConfigurations));
+        dependsOn(cluster.getPluginAndModuleConfigurations());
         getClusters().add(cluster);
     }
 
     default void useCluster(Provider<ElasticsearchCluster> cluster) {
         useCluster(cluster.get());
     }
 
-    default void beforeStart() {}
+    default void beforeStart() {
+        getTasksService().get().register(this);
```

Comment:
We need to do that here as we cannot apply custom logic within task graph calculation as this is skipped when configuration cache is enabled